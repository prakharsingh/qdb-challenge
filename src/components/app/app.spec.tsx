import * as React from 'react';
import { cleanup, waitFor } from '@testing-library/react';

import renderWithRouter from '../../test-utils/render-with-router';
import App from './app';

describe('<App/>', () => {
  afterEach(cleanup);

  it('SHOULD render successfully', async () => {
    const { asFragment, getByTestId } = renderWithRouter(App);
    const container = await waitFor(() => getByTestId('container'));

    expect(container.children).toHaveLength(2);
    expect(asFragment()).toMatchSnapshot();
  });
});
