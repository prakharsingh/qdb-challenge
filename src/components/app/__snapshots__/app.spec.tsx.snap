// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> SHOULD render successfully 1`] = `
<DocumentFragment>
  <section
    class="ant-layout ant-layout-has-sider sc-AxheI kAhrYl"
    data-testid="container"
  >
    <aside
      class="ant-layout-sider ant-layout-sider-dark"
      style="flex: 0 0 200px; max-width: 200px; min-width: 200px; width: 200px;"
    >
      <div
        class="ant-layout-sider-children"
      >
        <div
          class="ant-page-header ant-page-header-ghost"
        >
          <div
            class="ant-page-header-heading"
          >
            <div
              class="ant-page-header-heading-left"
            >
              <span
                class="ant-page-header-heading-title"
                title="QDB Challenge"
              >
                QDB Challenge
              </span>
            </div>
          </div>
        </div>
        <div
          class="ant-space ant-space-vertical sc-AxiKw hSLCUF"
        >
          <div
            class="ant-space-item"
            style="margin-bottom: 8px;"
          >
            <div
              class="ant-space ant-space-vertical sc-AxiKw hSLCUF"
            >
              <div
                class="ant-space-item"
                style="margin-bottom: 8px;"
              >
                <div
                  class="sc-AxjAm jWlVxK"
                >
                  <span
                    class="ant-avatar ant-avatar-circle ant-avatar-icon"
                    style="width: 128px; height: 128px; line-height: 128px; font-size: 64px;"
                  >
                    <span
                      aria-label="user"
                      class="anticon anticon-user"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        class=""
                        data-icon="user"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                        />
                      </svg>
                    </span>
                  </span>
                </div>
              </div>
              <div
                class="ant-space-item"
              >
                <div
                  class="sc-AxjAm jWlVxK"
                >
                  <div
                    data-testid="user-name"
                  />
                  <div
                    data-testid="user-email"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-space-item"
            style="margin-bottom: 8px;"
          >
            <div
              class="ant-divider ant-divider-horizontal"
              role="separator"
            />
          </div>
          <div
            class="ant-space-item"
          >
            <div
              class="sc-AxirZ cUVYWJ"
            >
              <ul
                class="ant-menu ant-menu-light ant-menu-root ant-menu-vertical"
                role="menu"
              >
                <li
                  class="ant-menu-item ant-menu-item-only-child"
                  role="menuitem"
                >
                  <a
                    href="/dashboard"
                  >
                    Dashboard
                  </a>
                </li>
                <li
                  class="ant-menu-item ant-menu-item-only-child"
                  role="menuitem"
                >
                  <a
                    href="/report"
                  >
                    Report
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </aside>
    <main
      class="ant-layout-content"
    >
      <div
        class="sc-AxhCb cJYvcU"
      >
        Dashboard View
      </div>
    </main>
  </section>
</DocumentFragment>
`;
